virttest/libvirt_xml/vm_xml.py

```diff
@@ -727,6 +727,22 @@ def check_cpu_mode(mode):
             raise xcepts.LibvirtXMLError(
                 "The cpu mode '%s' is invalid!" % mode)
 
+    def get_cpu_topology(self):
+        """
+        Return cpu topology dict
+        """
+        topology = {}
+        xmltreefile = self.__dict_get__('xml')
```

Comment:
topology is element of cpu xml, so you could get it with
vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name, virsh_instance=virsh_instance)
topology = vmxml.cpu.topology

as cpu could be without topology setting, so you need use try..except to return None if not found